doc/api/assert.markdown

```diff
@@ -39,27 +25,72 @@ non-enumerable:
     // WARNING: This does not throw an AssertionError!
     assert.deepEqual(Error('a'), Error('b'));
 
+## assert.deepStrictEqual(actual, expected[, message])
+
+Tests for deep equality. Primitive values are compared with the strict equality
+operator ( `===` ).
+
+## assert.doesNotThrow(block[, error][, message])
+
+Expects `block` not to throw an error. See [assert.throws()](#assert_assert_throws_block_error_message) for more details.
+
+If `block` throws an error and if it is of a different type from `error`, the
+thrown error will get propagated back to the caller. The following call will
+throw the `TypeError`, since we're not matching the error types in the
+assertion.
+
+    assert.doesNotThrow(
+      function() {
+        throw new TypeError("Wrong value");
+      },
+      SyntaxError
+    );
+
+In case `error` matches with the error thrown by `block`, an `AssertionError`
+is thrown instead.
+
+    assert.doesNotThrow(
+      function() {
+        throw new TypeError("Wrong value");
+      },
+      TypeError
+    );
+
+## assert.equal(actual, expected[, message])
+
+Tests shallow, coercive equality with the equal comparison operator ( `==` ).
+
+## assert.fail(actual, expected, message, operator)
+
+Throws an exception that displays the values for `actual` and `expected`
+separated by the provided operator.
+
+## assert.ifError(value)
+
+Throws `value` if `value` is truthy. This is useful when testing the `error`
+argument in callbacks.
+
 ## assert.notDeepEqual(actual, expected[, message])
 
 Tests for any deep inequality. Opposite of `assert.deepEqual`.
 
-## assert.strictEqual(actual, expected[, message])
+## assert.notDeepStrictEqual(actual, expected[, message])
```

Comment:
I wrestled with that myself, but I figured the initial PR should have it 100% alphabetical. Discussion would then steer it in either direction.

I am not in favor of one or the other. I can see the pros/cons of both; Having a reliable system for navigating the reference, and of having async/Sync being together.